Tidy up home page templates (Fixes #10871)

alexgibson opened the pull request:
## Description
- Moves/renames `mozorg/contentful-homepage.html` to `mozorg/home/hone-en.html`.
- Falls back to `mozorg/home/home.html` if Contentful errors.
- Creates a new `mozorg/home/base.html` template that all home pages extend from, so we avoid repeating metadata.
- Renames `home-2018` CSS bundles to `home-eu` since it relates only to the DE/FR home pages.
- Removes old MR1 promo from `mozorg/home/home.html`.
- Updates functional tests for home pages.

## Issue / Bugzilla link
#10871

## Testing
- http://localhost:8000/es-US/
- http://localhost:8000/de/
- http://localhost:8000/fr/
- http://localhost:8000/es-ES/

Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/424328544

maureenlholland reviewed (approved):
LGTM 🧹 
One question re: `home-en` name now that I see it in practice, not blocking

alexgibson commented on bedrock/mozorg/views.py:
```diff
@@ -122,27 +122,25 @@ def home_view(request):
         if switch("contentful-homepage-de"):
             try:
-                template_name = "mozorg/contentful-homepage.html"
+                template_name = "mozorg/home/home-en.html"
```
I think this is a good suggestion, it will also match the CSS bundle name then 👍

alexgibson commented:
@maureenlholland thanks for the review, updated. I also removed the (currently unused) experimental home page URL at https://www.mozilla.org/en-US/exp/, as even right now this is showing the wrong home page in production.

maureenlholland commented on bedrock/firefox/redirects.py:
```diff
@@ -547,4 +547,6 @@ def firefox_channel(*args, **kwargs):
     # issue 10703
     redirect(r"firefox/lockwise/?", "https://support.mozilla.org/kb/end-of-support-firefox-lockwise"),
+    # issue 10879
+    redirect(r"^/exp/?$", "mozorg.home"),
```
👍

maureenlholland reviewed (approved):
🙌  nice catch on /exp too. r++
